extension/chrome/elements/compose.ts

```diff
@@ -205,6 +215,12 @@ export class ComposeView extends View {
     this.S.cached('body').on('focusin', setActiveWindow);
     this.S.cached('body').on('click', setActiveWindow);
     this.S.cached('icon_help').on('click', this.setHandler(async () => await this.renderModule.openSettingsWithDialog('help'), this.errModule.handle(`help dialog`)));
+    this.S.cached('input_text').on('paste', this.setHandler(async (el, ev: JQuery.Event) => {
```

Comment:
FYI `S.cached('input_text')` and `S.cached('input_intro')` may work differently because they are defined differently in `extension/chrome/elements/compose.ts`
```
public S = Ui.buildJquerySels({
...
input_text: 'div#input_text',
...
input_intro: '.input_intro',
...
```

Also, please remove `async` from handler function definitions: they are unnecessary and confusing.

Reply:
Thanks for remarks. Both contenteditable div are being selected successfully and the `onpaste` event gets triggered but I have no clue why it won't work for `input_text`. I also tried binding the `onpaste` event on its child node but still doesn't give the expected result while `input_intro` works very well.

Reply:
I started to get more info with the issue. The `ev.currentTarget` and `ev.target` is different when the event is triggered . The currentTarget is `div#input_text` which is correct, however, the `ev.target` is either the child node within the currentTarget and that is `<div>` and `<br>` so the paste being prevented is only for that two element making the `ev.preventDefault()` ineffective.

Reply:
Oh you mean after these appends:
```
Xss.sanitizeAppend(this.view.S.cached('input_text'), `<br/>\n<br/>\n<br/>\n${Xss.escape(e.data)}`);
```
and
```
Xss.sanitizeAppend(this.view.S.cached('input_text'), this.getTripleDotSanitizedFormattedHtmlContent());
```

Have you tried to use `#input_text` instead of `div#input_text`? Would that help?
That is, add one more mapping like
```
public S = Ui.buildJquerySels({
...
input_text_full: '#input_text',
```
and then attach the handler
` this.S.cached('input_text_full').on('paste' ... `

Reply:
> Have you tried to use #input_text instead of div#input_text? Would that help?

Yes, I do but unfortunately, it doesn't work.

Reply:
Here is the comparison for an event on `input_text` vs `input_intro`:

`input_text`:
![image](https://user-images.githubusercontent.com/46025304/210219638-146e3b67-4dcd-47c8-8f2d-271eac96f412.png)

`input_intro`:
![image](https://user-images.githubusercontent.com/46025304/210219893-479ef5d1-a3c1-4c55-9a9a-8ac8eb5ca285.png)

We can notice that input_intro has `currentTarget` and `target` has the same value. Where in input_text, `currentTarget` has the right element and `target` was set to `br` element.

When it comes to element hierarchy, `div#input_text` is the parent and the one should be prioritize to received the paste event over the `br` element?

![image](https://user-images.githubusercontent.com/46025304/210220280-b6b4819a-5b9b-49bb-8d76-a024078294a6.png)

Reply:
Technically the event was generated at `br` level.
Can you tried something like https://stackoverflow.com/a/10558535 ?

Reply:
During my testing case with somehow blocking child nodes from receiving the event, it doesn't work. `currentTarget: div#input_text` while `target: br`.

Reply:
I'd also like to input limits on public key import, do you have any recommendation of what would be a good number for the input? Do you think it would reach 5Kb?

Reply:
> During my testing case with somehow blocking child nodes from receiving the event, it doesn't work. `currentTarget: div#input_text` while `target: br`.

Have you played with this fiddle http://jsfiddle.net/xvKNC/2/   
?
Can you create a fiddle setup for the problem?
And test how it works with `.on("paste", "br" ...` (div2 example)

Reply:
> I'd also like to input limits on public key import, do you have any recommendation of what would be a good number for the input? Do you think it would reach 5Kb?

If there is a single key, probably not. Need to check whether there is `parse` or `parseMany` etc.

Reply:
@rrrooommmaaa I tried to do simple experiment on jsfiddle to conclude if try to prevent paste with `onpaste` here :https://jsfiddle.net/zqfx2h14/ and all contenteditable obeys the ev.preventdefault. 

on the extension, it only prevents pasting on input_intro but not on input_text. We can remember that input_intro doesn't have any change in terms of functionality unlike the input_text. I think the data has been processed/injected on <div> in div#input_text that's why binding paste don't really achieve anything there.

Reply:
It seems i found the right place: https://github.com/FlowCrypt/flowcrypt-browser/blob/43255ea7bf508f02fff685693a2e1460f1fb690d/extension/chrome/elements/compose-modules/compose-input-module.ts#L81

Reply:
> I tried to do simple experiment on jsfiddle to conclude if try to prevent paste with `onpaste` here :https://jsfiddle.net/zqfx2h14/ and all contenteditable obeys the ev.preventdefault.
> 
> on the extension, it only prevents pasting on input_intro but not on input_text. We can remember that input_intro doesn't have any change in terms of functionality unlike the input_text. I think the data has been processed/injected on
> 
> in div#input_text that's why binding paste don't really achieve anything there.

There can be `<br/>`s added programmatically, so for fiddling, I think, we'll need something like this:
```
<div contenteditable style="border: 1px solid orange">something before<br/>something after</div>
```